api/runtime/hooks/v1alpha1/inplaceupdate_types.go

```diff
@@ -102,30 +102,32 @@ type CanUpdateMachineSetRequest struct {
 	// CommonRequest contains fields common to all request types.
 	CommonRequest `json:",inline"`
 
-	// current contains the current MachineSet spec and related template objects.
+	// current contains the current state of the MachineSet and related objects.
 	// +required
-	Current CanUpdateMachineSetRequestObjects `json:"current"`
+	Current CanUpdateMachineSetRequestObjects `json:"current,omitempty,omitzero"`
 
-	// desired contains the desired MachineSet spec and related template objects.
+	// desired contains the desired state of the MachineSet and related objects.
 	// +required
-	Desired CanUpdateMachineSetRequestObjects `json:"desired"`
+	Desired CanUpdateMachineSetRequestObjects `json:"desired,omitempty,omitzero"`
 }
 
 // CanUpdateMachineSetRequestObjects groups objects for CanUpdateMachineSetRequest.
 type CanUpdateMachineSetRequestObjects struct {
-	// machineSetSpec is the MachineSet spec.
+	// machine is the full MachineSet object.
 	// +required
-	MachineSetSpec clusterv1beta1.MachineSetSpec `json:"machineSetSpec"`
+	MachineSet clusterv1.MachineSet `json:"machineSet,omitempty,omitzero"`
```

Comment:
I think we should also use the full objects here (but also here we won't send status)